### ISSUE TITLE
UndocumentedPublicClass false positive with annotations

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#194]
The following code:

```kotlin
/**
 * @author niklasbaudy / vanniktech
 * @since 07.07.17
 */
@Module class BluetoothModule {
  // Some Dagger 2 methods
}
```

gets flagged: `UndocumentedPublicClass - [<UnknownName>] at src/main/java/my/package/android/bluetooth/BluetoothModule.kt:21:41`

**Comment by arturbosch** [comment]
Will test it ^^

**Comment by arturbosch** [comment]
My test case works oO

```kotlin
	it("should not report when annotations are between class keyword") {
		val code = """
			/**
			 * @author niklasbaudy / vanniktech
			 * @since 07.07.17
			 */
			@Module class BluetoothModule {
			  // Some Dagger 2 methods
			}
		"""

		assertThat(subject.lint(code)).isEmpty()
	}
```

**Comment by arturbosch** [comment]
Maybe send me the whole class via email or a link if its on GitHub.

`<UnknownName>` shouldn't appear for classes oO
```kotlin
internal fun PsiElement.searchName(): String {
	return this.namedUnwrappedElement?.name ?: "<UnknownName>"
}
```

**Comment by vanniktech** [comment]
That's the entire file:

```kotlin
package my.package.android.bluetooth

import android.app.Application
import dagger.Module
import dagger.Provides
import my.package.bluetoothsdk.Logger
import my.package.bluetoothsdk.RealRxBluetooth
import my.package.bluetoothsdk.RxBluetooth
import timber.log.Timber
import javax.inject.Singleton

/**
 * <br>
 * FAMILO GmbH
 *
 * @author niklasbaudy / vanniktech
 * @since 07.07.17
 */
@Module class BluetoothModule {
  @Provides @Singleton internal fun provideRxPermissions(application: Application): RxBluetooth {
    return RealRxBluetooth(application, object : Logger {
      override fun d(message: String) {
        Timber.d(message)
      }
    })
  }
}
```

And the error message:

```
UndocumentedPublicClass - [<UnknownName>] at src/main/java/my/package/android/bluetooth/BluetoothModule.kt:21:41
```

**Comment by arturbosch** [comment]
Ok I can confirm via test case that something is wrong here :)

**Comment by arturbosch** [comment]
Can I use your code in a test case?

**Comment by vanniktech** [comment]
Sure. Knock yourself out.

**Comment by arturbosch** [comment]
Ha, interestingly this is a false positive for the real reason in #213  :). It is fixed https://github.com/arturbosch/detekt/pull/221

**Comment by vanniktech** [comment]
Cool thanks :)
